### PR TITLE
Add ability to define the amount of retries

### PR DESCRIPTION
**Pull request by rickycheers**

### CODE REVIEW

**Comment by rickycheers** [comment]
Hi there!

I recently came across a use case where I needed to "turn off" or "disable" the automatic retries to prevent additional requests from flooding our web servers.

The way I worked around it was by manually creating the Guzzle client myself instead of using the this factory class. I figured something like this would be useful by someone else, so here it is. It not only allows you to disable the retries, but it even lets you increment them if needed.

Hope you find this useful, let me know if you have any comments.

**Comment by GrahamCampbell** [comment]
Thanks for the PR. I don't follow the use case, however. If you don't need retries, then you don't need this package at all. Why can't you just do `$client = new Client()`?

**Comment by GrahamCampbell** [comment]
Ah, I see, you want to easily turn it off temporarily.

File: src/GuzzleFactory.php [hunk 1]
```diff
@@ -81,14 +87,19 @@ public static function make(array $options = [], int $backoff = null, array $cod
      * @param int|null                      $backoff
      * @param int[]|null                    $codes
      * @param \GuzzleHttp\HandlerStack|null $stack
+     * @param int|null                      $retries
      *
      * @return \GuzzleHttp\HandlerStack
      */
-    public static function handler(int $backoff = null, array $codes = null, HandlerStack $stack = null)
+    public static function handler(int $backoff = null, array $codes = null, HandlerStack $stack = null, int $retries = null)
```
Comment (GrahamCampbell) [inline src/GuzzleFactory.php]: I'd prefer the parameter order to have stack at the end. We can just make this a major version bump.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by rickycheers** [comment]
> Ah, I see, you want to easily turn it off temporarily.

Yeah, either that or simply be able to configure how many additional attempts to perform. I can imagine folks wanting to just make one additional attempt instead of 3

**Comment by GrahamCampbell** [comment]
Thanks for working on this, and sorry for my slow merge! 🍻

**Comment by GrahamCampbell** [comment]
v5.0.0 is out now. 🚀

**Pull request merged**